tests/test_item.py

```diff
@@ -0,0 +1,187 @@
+            item_id=uuid.uuid4(),
+            name='cubo',
+            price=15,
+            description='desc2'
+        )
+
+        resp = self.app.get('/items/')
+        assert resp.status_code == OK
+        assert json.loads(resp.data.decode()) == [obj1.json(), obj2.json()]
+
+    def test_post_create_item__success(self):
+        source_item = {
+            'name': 'cubo',
+            'price': 15,
+            'description': 'desc1'
+        }
+
+        resp = self.app.post('/items/', data=source_item)
+        assert resp.status_code == CREATED
+        resp2 = self.app.get('/items/')
+
+        db_data = {
+            'name': json.loads(resp2.data.decode())[0]['name'],
+            'price': json.loads(resp2.data.decode())[0]['price'],
+            'description': json.loads(resp2.data.decode())[0]['description']
+        }
+
+        assert source_item == db_data
+
+    def test_post__malformed_input_type(self):
+        source_item = {
+            'name': 'cubo',
+            'price': 'banana',
+            'description': 'desc1'
+        }
+
+        source_item_2 = {
+            'name': 'rombo',
+            'description': 'desc2'
+        }
+        resp = self.app.post('/items/', data=source_item)
+        assert resp.status_code == BAD_REQUEST
+        resp = self.app.post('/items/', data=source_item_2)
+        assert resp.status_code == BAD_REQUEST
+
+    def test_get_item__has_found(self):
+        obj1 = Item.create(
+            item_id=uuid.uuid4(),
+            name='cubo',
+            price=5,
+            description='dhfsdjofgjasdogj'
+        )
+
+        resp = self.app.get('/item/{}'.format(obj1.item_id))
+        assert resp.status_code == OK
+
+        item = json.loads(resp.data.decode())
+        assert item == obj1.json()
+
+    def test_get_item__not_found(self):
+
+        resp = self.app.get('/item/{}'.format(uuid.uuid4()))
+        assert resp.status_code == NOT_FOUND
+
+    def test_item__deleted_successfully(self):
+        obj1 = Item.create(
+            item_id=uuid.uuid4(),
+            name='cubo',
+            price=5,
+            description='dhfsdjòfgjasdògj'
+        )
+
+        obj2 = Item.create(
+            item_id=uuid.uuid4(),
+            name='triangolo',
+            price=10,
+            description='Roba a caso, totally random'
+        )
+        resp = self.app.delete('item/{}'.format(obj1.item_id))
+        assert resp.status_code == NO_CONTENT
+        assert len(Item.select()) == 1
+        resp = self.app.get('item/{}'.format(obj1.item_id))
+        assert resp.status_code == NOT_FOUND
+        assert Item.get(item_id=obj2.item_id)
+
+    def test_item__delete_not_found(self):
+        Item.create(
+            item_id=uuid.uuid4(),
+            name='cubo',
+            price=5,
+            description='dhfsdjòfgjasdògj'
+        )
+
+        resp = self.app.delete('item/{}'.format(uuid.uuid4()))
+        assert resp.status_code == NOT_FOUND
```

Comment:
a questo punto verifichiamo che l'item creato esista sempre (o rimuoviamo la creazione dell'Item, che in questo momento è inutile)